ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -1010,35 +1047,34 @@ public void write() throws StoreException {
       try {
         if (indexes.size() > 0) {
           // before iterating the map, get the current file end pointer
-          Map.Entry<Long, IndexSegment> lastEntry = indexes.lastEntry();
+          Map.Entry<Offset, IndexSegment> lastEntry = indexes.lastEntry();
           IndexSegment currentInfo = lastEntry.getValue();
-          long currentIndexEndOffsetBeforeFlush = currentInfo.getEndOffset();
-          long logEndOffsetBeforeFlush = logSegment.getEndOffset();
-          if (logEndOffsetBeforeFlush < currentIndexEndOffsetBeforeFlush) {
+          Offset currentIndexEndOffsetBeforeFlush = currentInfo.getEndOffset();
```

Comment:
Actually, renaming this would make it inconsistent with how this is referred to in other places (`BlobStore.java`). I don't think the name is misleading either - any reason to suggest the change?